variable-for-property rule should allow reserved words

**Issue**
The rule `variable-for-property` reports reserved words as being non-variables (which is technically true of course). This means it's impossible to use words like `transparent` or `none`.

**Possible solution**
Make a list of reserved words that are allowed when using `variable-for-property` (equal to [scss-lint](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#variableforproperty), see [this line](https://github.com/brigade/scss-lint/blob/e9a8e243b43bc4fe30a919ebea2e413087546a95/lib/scss_lint/linter/variable_for_property.rb#L6)).

**Example**
This can be tested with the following config:

```yml
options:
  merge-default-rules: false
  formatter: stylish
rules:
  variable-for-property:
    - 2
    -
      properties:
        - background
```

And this CSS:

```css
.test {
  background: inherit;
  background: initial;
  background: transparent;
  background: none;
}
```

Which generates this errors when using sass-lint v1.5.1:

```
test.scss
  2:3  error  Values for properties of type 'background' may only be variables  variable-for-property
  3:3  error  Values for properties of type 'background' may only be variables  variable-for-property
  4:3  error  Values for properties of type 'background' may only be variables  variable-for-property
  5:3  error  Values for properties of type 'background' may only be variables  variable-for-property

✖ 4 problems (4 errors, 0 warnings)
```

--- a/docs/rules/variable-for-property.md
+++ b/docs/rules/variable-for-property.md
@@ -1,6 +1,12 @@
 # Variable For Property
 
-Rule `variable-for-property` will enforce the use of variables for the values of specified properties. There are no properties by default.
+Rule `variable-for-property` will enforce the use of variables for the values of specified properties.
+There are no properties by default, except for reserved words listed below which are always whitelisted:
+* inherit
+* initial
+* transparent
+* none
+* currentColor
 
 ## Options
 

--- a/lib/rules/variable-for-property.js
+++ b/lib/rules/variable-for-property.js
@@ -2,6 +2,27 @@
 
 var helpers = require('../helpers');
 
+// The whitelisted ident values
+var whitelistedValues = ['inherit', 'initial', 'transparent', 'none', 'currentColor'];
+
+/**
+ * Checks If the property is of a valid type, either its a variable or it's a whitelisted ident value
+ *
+ * @param {Object} propertyElem - The property element
+ * @returns {boolean} Whether the property is valid or not
+ */
+var isValidProperty = function (propertyElem) {
+  if (propertyElem) {
+    if (propertyElem.type === 'variable') {
+      return true;
+    }
+    else if (propertyElem.type === 'ident' && whitelistedValues.indexOf(propertyElem.content) !== -1) {
+      return true;
+    }
+  }
+  return false;
+};
+
 module.exports = {
   'name': 'variable-for-property',
   'defaults': {
@@ -19,7 +40,7 @@ module.exports = {
         if (declarationType === 'ident') {
           if (parser.options.properties.indexOf(declarationIdent) !== -1) {
             node.forEach(function (valElem) {
-              if (valElem.type !== 'variable') {
+              if (!isValidProperty(valElem)) {
                 result = helpers.addUnique(result, {
                   'ruleId': parser.rule.name,
                   'line': declaration.start.line,

--- a/tests/rules/variable-for-property.js
+++ b/tests/rules/variable-for-property.js
@@ -24,7 +24,8 @@ describe('variable for property - scss', function () {
         {
           'properties': [
             'margin',
-            'content'
+            'content',
+            'background'
           ]
         }
       ]
@@ -57,7 +58,8 @@ describe('variable for property - sass', function () {
         {
           'properties': [
             'margin',
-            'content'
+            'content',
+            'background'
           ]
         }
       ]

--- a/tests/sass/variable-for-property.sass
+++ b/tests/sass/variable-for-property.sass
@@ -4,7 +4,7 @@
 
   &__element
     margin: $margin
-  
+
 
 
 =blue()
@@ -17,8 +17,14 @@
 
   &__element
     margin: 0
-  
+
 
 
 =red()
   margin: 0
+
+.test
+  background: inherit
+  background: initial
+  background: transparent
+  background: none

--- a/tests/sass/variable-for-property.scss
+++ b/tests/sass/variable-for-property.scss
@@ -23,3 +23,10 @@
 @mixin red() {
   margin: 0;
 }
+
+.test {
+  background: inherit;
+  background: initial;
+  background: transparent;
+  background: none;
+}

